### ISSUE TITLE
generating so many refresh tokens

### ISSUE THREAD

**Issue by Glyphack** [flavors/django-graphql-jwt#81]
Hi, I'm using your library on my app there is something I wanted to ask you:
For every refereshToken mutation I get a new refreshToken and and a new token, the refreshToken is saved to database and the old refreshToken which was used to renew refreshToken and token will become revoked. In my opinion this is so much unnecessarry INSERT and UPDATE for the db. maybe you should not renew refreshToken every time we use refreshToken mutation and just renew it a time very close to expire date(like a day before), in this way you can reduce db operations.

**Comment by mongkok** [comment]
Hi @Glyphack !
The refresh token is not revoked when the `RefreshToken` mutation is called.

If you want to revoke it, you can call to `Revoke` mutation or create a listener for `refresh_token_rotated` signal (you can also delete it).

In my opinion, clients and not the API should decide when to refresh the token.

**Comment by Glyphack** [comment]
I did used refresh_token_rotated.
about the last sentence I agree with you but there is something:

> For every refereshToken mutation I get a new refreshToken and and a new token
So you don't have any choices other than getting a new refreshToken when you want to get a new Token(accessToken), and every refreshToken is being saved on db.

**Comment by mongkok** [comment]
This is a query to get a new JWT `token` without creating a new `refreashToken`.

```graphql
mutation RefreshToken($refreshToken: String!) {
  refreshToken(refreshToken: $refreshToken) {
    token
    payload
  }
}
```
Until the last release v0.2.0, a `refreshToken` was created even if it is not requested in the query.
This problem was fixed [PR #77](https://github.com/flavors/django-graphql-jwt/pull/77), I hope to create a new release this week.

**Comment by ryarasi** [comment]
If anyone stumbles upon this, I found a way to configure the reuse of refreshtokens so that new records are not created every time refreshtoken is requested.

```
GRAPHQL_JWT = {
    "JWT_VERIFY_EXPIRATION": True,
    "JWT_LONG_RUNNING_REFRESH_TOKEN": True,
    'JWT_REUSE_REFRESH_TOKENS': True, # Eliminates creation of new db records every time refreshtoken is requested.
    'JWT_EXPIRATION_DELTA': timedelta(minutes=60), # Expiry time of token
    'JWT_REFRESH_EXPIRATION_DELTA': timedelta(days=7), # Expiry time of refreshToken
}
```

[Source](https://readthedocs.org/projects/django-graphql-jwt/downloads/pdf/latest/)
